WBooks/MainMenu/MainMenuController.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  MainMenuController.swift
+//  WBooks
+//
+//  Created by Megan Cole on 14/01/2020.
+//  Copyright © 2020 Wolox. All rights reserved.
+//
+
+import UIKit
+import WolmoCore
+
+class MainMenuController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    private let _view: MainMenuView = MainMenuView.loadFromNib()!
+    var booksArray: [Book] = []
+    
+    func tableView(_ booksTable: UITableView, numberOfRowsInSection section: Int) -> Int {
+        booksArray.count
+    }
+
+    func tableView(_ booksTable: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = _view.booksTable.dequeue(cell: BookCell.self)!
```

Comment:
Forced unwrapping (the exclamation mark) may lead to unexpected errors, so let's use safe unwrapping instead 😉 
You can use `guard` if you want. Here's an example 
```Swift
guard let cell = _view.booksTable.dequeue(cell: BookCell.self) else { return UITableViewCell() }
```

Reply:
Apart from that, you can also make the table reuse cells in order to save memory. Here are the steps:

1. Add a cell identifier in `MainMenuController`. 
```Swift
private static let cellIdentifier = "BookCell"
```

2. Change `register` so that it uses the cell identifier: 
```Swift
_view.booksTable.register(UINib(nibName: "BookCell", bundle: nil), forCellReuseIdentifier: MainMenuController.cellIdentifier)
```

3. Use `dequeueReusableCell` instead of `dequeue`:
```Swift
guard let cell = _view.booksTable.dequeueReusableCell(withIdentifier: MainMenuController.cellIdentifier) as? BookCell else { return UITableViewCell() }
```

Reply:
great suggestion! thanks